hr_worked_days_from_timesheet/tests/test_worked_days_from_timesheet.py

```diff
@@ -0,0 +1,160 @@
+            'name': 'ddd',
+            'account_id': self.analytic.id,
+            'unit_amount': 3.0,
+            'journal_id': self.analytic_journal.id,
+        }
+        self.analytic_ts1 = self.env['hr.analytic.timesheet'].with_context(
+            user_id=self.user_test.id).create(
+                analytic_ts1_dict)
+
+        analytic_ts2_dict = {
+            'date': '2014-01-02',
+            'user_id': self.user_test.id,
+            'employee_id': self.employee.id,
+            'name': 'ddd',
+            'account_id': self.analytic.id,
+            'unit_amount': 5.0,
+            'journal_id': self.analytic_journal.id,
+            'sheet_id': self.ts2.id,
+        }
+        self.analytic_ts2 = self.env['hr.analytic.timesheet'].with_context(
+            user_id=self.user_test.id).create(
+                analytic_ts2_dict)
+
+        analytic_ts3_dict = {
+            'date': '2014-01-04',
+            'user_id': self.user_test.id,
+            'sheet_id': self.ts2.id,
+            'name': 'ddd',
+            'account_id': self.analytic.id,
+            'unit_amount': 7.0,
+            'journal_id': self.analytic_journal.id,
+        }
+        self.analytic_ts3 = self.env['hr.analytic.timesheet'].with_context(
+            user_id=self.user_test.id).create(
+                analytic_ts3_dict)
+
+        analytic_ts4_dict = {
+            'date': '2014-01-15',
+            'user_id': self.user_test.id,
+            'sheet_id': self.ts4.id,
+            'name': 'ddd',
+            'account_id': self.analytic.id,
+            'unit_amount': 11.5,
+            'journal_id': self.analytic_journal.id,
+        }
+        self.analytic_ts4 = self.env['hr.analytic.timesheet'].with_context(
+            user_id=self.user_test.id).create(
+                analytic_ts4_dict)
+
+        analytic_ts5_dict = {
+            'date': '2014-01-16',
+            'user_id': self.user_test.id,
+            'sheet_id': self.ts1.id,
+            'name': 'ddd',
+            'account_id': self.analytic.id,
+            'unit_amount': 13,
+            'journal_id': self.analytic_journal.id,
+        }
+        self.analytic_ts5 = self.env['hr.analytic.timesheet'].with_context(
+            user_id=self.user_test.id).create(
+                analytic_ts5_dict)
+
+        ts = self.ts1 + self.ts2 + self.ts3 + self.ts4
+        ts.button_confirm()
+        ts.signal_workflow('done')
+
+    def test_timesheet_import(self):
+        payslip_dict = {
+            'employee_id': self.employee.id,
+            'contract_id': self.contract.id,
+            'date_from': '2014-01-02',
+            'date_to': '2014-01-15',
+        }
+        payslip = self.env['hr.payslip'].create(
+            payslip_dict)
+        payslip.import_worked_days()
+        for wd in payslip.worked_days_line_ids:
+            self.assertIn(wd.number_of_hours, [11.5, 12])
+
+        payslip.import_worked_days()
+        self.assertEqual(len(payslip.worked_days_line_ids), 2)
```

Comment:
Some unit tests could be added to test the date intervals when importing timesheets.